### PR TITLE
[#48219] Implement global index page for meetings

### PR DESCRIPTION
**Pull request by klaustopher**
https://community.openproject.org/projects/openproject/work_packages/48219

### CODE REVIEW

File: modules/meeting/lib/open_project/meeting/engine.rb [hunk 1]
```diff
@@ -63,6 +63,17 @@ class Engine < ::Rails::Engine
...
+      #       menu :top_menu,
+      #            :meetings, { controller: '/meetings', project_id: nil, action: 'index' },
+      #            context: :modules,
+      #            caption: I18n.t('label_meetings_plural'),
+      #            if: Proc.new {
+      #              (User.current.logged? || !Setting.login_required?) &&
+      #                User.current.allowed_to_globally?(:view_meetings)
+      #            }
+      #     end
+
```
Comment (klaustopher) [inline modules/meeting/lib/open_project/meeting/engine.rb]: I guess one option would be to add the check for the feature flag into the `if:` proc. That would be fine as well. But if anybody has another suggestion, please let me know.

File: config/initializers/menus.rb [hunk 2]
```diff
@@ -54,6 +54,21 @@
               (User.current.logged? || !Setting.login_required?) &&
                 User.current.allowed_to_globally?(:view_news)
             }
+
+  # TODO: This should be put in modules/meeting/../engine.rb, but if it is put
+  # in there, it causes a NameError because `Setting` that is used by FeatureDecisions
+  # is not loaded yet. So, let's see what to do here...
+  if OpenProject::FeatureDecisions.more_global_index_pages_active?
```
Comment (oliverguenther) [inline config/initializers/menus.rb]: You can move the if into the `if:` Proc below, that means the menu is registered in code but never loaded.

Comment (klaustopher) [reply config/initializers/menus.rb]: Yeah, wrote that in https://github.com/opf/openproject/pull/12641#discussion_r1196372340. I'll refactor to do that

File: modules/meeting/lib/open_project/meeting/engine.rb (hunk 1 above)
Comment (oliverguenther) [reply modules/meeting/lib/open_project/meeting/engine.rb]: We could look into statically loading the feature decisions, but as it depends on `Settings`, this is not part of the initialization autoloading as that changed with Rails 7. I don't have a quick fix for that, so would vote for moving it into the proc.

File: config/initializers/menus.rb (hunk 2 above)
Comment (klaustopher) [reply config/initializers/menus.rb]: Done ✅

**Review by oliverguenther — APPROVED** [review]
LGTM, feel free to move the feature decision check into the proc and then merge it.

**Pull request merged**